main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -213,15 +213,31 @@ object Parser2 {
     stack.head
   }
 
+  /**
+    * Get first non-comment previous position of the parser as a [[SourceLocation]].
+    * TODO: It might make sense to seek the first non-comment position here.
+    */
+  private def previousSourceLocation()(implicit s: State): SourceLocation = {
```

Comment:
I am confused. The token is the previous one, but the source location is similar to currentSourceLocation. What gives?